src/mosquito/redis_backend.cr

```diff
@@ -124,14 +124,14 @@ module Mosquito
       key = build_key(LIST_OF_QUEUES_KEY)
       list_queues = redis.zrange(key, 0, -1).as(Array)
 
-      return [] of String unless list_queues.any?
+      return [] of String if list_queues.empty?
 
       list_queues.compact_map(&.as(String))
     end
 
     def self.list_runners : Array(String)
-      runner_prefix = "mosquito:runners:"
-      Redis.instance.keys("#{runner_prefix}*")
+      runner_prefix = build_key(LIST_OF_QUEUES_KEY)
+      Redis.instance.keys("#{runner_prefix}:*")
```

Comment:
Nice improvement. The use of #keys here is still a performance implication and can abuse redis servers regardless of the search string, but that's an unrelated issue.

Reply:
Perhaps it's possible to use build_key once here, and then cache the value in the getter `runner_prefix` to improve performance, but I'm not sure.